### PR TITLE
Check namings in async statements

### LINKED ISSUE #80: Support `async` functions and method
Currently this extensions does not support `AsyncFunctionDef`.
And does not check anything in it unlike in normal `FunctionDef`.

I guess this should be addressed, since many people use `async` and it is a part of the language.

### PR DESCRIPTION
**Pull request by 5j9**
Fixes #80

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,6 +17,10 @@ matrix:
       python: 3.5
     - env: TOXENV=py36
       python: 3.6
+    - env: TOXENV=py37
+      dist: xenial
+      sudo: required
+      python: 3.7
```
Comment (sigmavirus24) [inline .travis.yml]: Should we also test against `nightly` on xenial too and allow for it to fail? Maybe something for another PR

File: src/pep8ext_naming.py [hunk 2]
```diff
@@ -14,11 +14,24 @@
...
-METHOD_CONTAINER_NODES = {ast.If, ast.While, ast.For, ast.With} | (
-    {ast.TryExcept, ast.TryFinally} if PY2 else {ast.Try})
+METHOD_CONTAINER_NODES = {ast.If, ast.While, ast.For, ast.With}
+
+if PY2:
+    METHOD_CONTAINER_NODES |= {ast.TryExcept, ast.TryFinally}
+else:
+    METHOD_CONTAINER_NODES |= {ast.Try}
+
+if PYTHON_VERSION > (3, 5):
```
Comment (sigmavirus24) [inline src/pep8ext_naming.py]: One nit - I'd rather have

```py
FUNC = (ast.FunctionDef,)
if PYTHON_VERSION > (3, 5):
    FUNC += (ast.AsyncFunctionDef,)
    METHOD_CONTAINER_NODES |= {...}
```

File: .travis.yml (hunk 1 above)
Comment (5j9) [reply .travis.yml]: That would be nice, but let's leave it for another PR.

File: src/pep8ext_naming.py (hunk 2 above)
Comment (5j9) [reply src/pep8ext_naming.py]: Done.

File: src/pep8ext_naming.py [hunk 3]
```diff
@@ -14,12 +14,21 @@
...
 
-PY2 = sys.version_info[0] == 2
+PYTHON_VERSION = sys.version_info[:3]
+PY2 = PYTHON_VERSION[0] == 2
 
 # Node types which may contain class methods
-METHOD_CONTAINER_NODES = {ast.If, ast.While, ast.For, ast.With} | (
-    {ast.TryExcept, ast.TryFinally} if PY2 else {ast.Try})
+METHOD_CONTAINER_NODES = {ast.If, ast.While, ast.For, ast.With}
+FUNC = (ast.FunctionDef,)
```
Comment (jparise) [inline src/pep8ext_naming.py]: `FUNC_NODES` for consistency?

Comment (5j9) [reply src/pep8ext_naming.py]: Done.

**Comment by 5j9** [comment]
> 
> 
> @5j9 hi, how's the progress going? 🙂
> I am blocked by this issue in my project: [wemake-services/wemake-python-styleguide#308](https://github.com/wemake-services/wemake-python-styleguide/issues/308)
> 
> Would be very happy about a new `pep8-naming` release.
> Is there's something I can help with?
> 
> Cheers!

@sobolevn , hi.
Unfortunately I neither have access nor much experiences with the release process. It's up to @sigmavirus24  and @florentx .

**Pull request merged**